book/src/building/swift-package/README.md

```diff
@@ -0,0 +1,221 @@
+```
+
+### Rust project setup
+
+```bash
+cargo new my_rust_lib --lib
+cd my_rust_lib
+```
+
+```toml
+# my_rust_lib/Cargo.toml
+
+[lib]
+crate-type = ["staticlib"]
+
+[build-dependencies]
+swift-bridge-build = "0.1"
+
+[dependencies]
+swift-bridge = "0.1"
+```
+
+In `lib.rs`, add the following:
+
+```rust
+// my_rust_lib/src/lib.rs
+#[swift_bridge::bridge]
+mod ffi {
+    extern "Rust" {
+        fn hello_rust();
+    }
+}
+
+fn hello_rust() -> String {
+    String::from("Hello Rust!")
+}
+```
+
+Add a new `build.rs` file:
+```rust
+// my_rust_lib/build.rs
+
+use std::path::PathBuf;
+
+fn main() {
+    let out_dir = PathBuf::from("./generated");
+
+    let bridges = vec!["src/lib.rs"];
+    for path in &bridges {
+        println!("cargo:rerun-if-changed={}", path);
+    }
+
+    swift_bridge_build::parse_bridges(bridges)
+        .write_all_concatenated(out_dir, env!("CARGO_PKG_NAME"));
+}
+```
+
+Build the project for the desired platforms:
+
+```bash
+cargo build --target x86_64-apple-darwin
+cargo build --target aarch64-apple-ios
+cargo build --target x86_64-apple-ios
+```
+
+## Creating the XCFramework
+
+Go back to the root of the project and make a new directory.
+
+```bash
+mkdir MyFramework && cd $_
+```
+
+Copy the generated libraries and the headers to this folder:
+
+```
+MyFramework
+├── include
+│   ├── SwiftBridgeCore.h
+│   ├── module.modulemap
+│   └── my_rust_lib.h
+├── ios
+│   └── libmy_rust_lib.a
+├── macos
+│   └── libmy_rust_lib.a
+└── simulator
+    └── libmy_rust_lib.a
+```
+
+Add `include/module.modulemap`:
+
+```modulemap
+module MyRustLib {
+    header "my_rust_lib.h"
+    header "SwiftBridgeCore.h"
+    export *
+}
+```
+
+Now it is time to build the xcframework:
+
+```bash
+xcodebuild -create-xcframework \
+    -library simulator/libmy_rust_lib.a \
+    -headers include \
+    -library ios/libmy_rust_lib.a \
+    -headers include \
+    -library macos/libmy_rust_lib.a \
+    -headers include \
+    -output MyRustLib.xcframework
+```
+
+*The order of the `library` tags is important*
+
+## Creating the Swift package
+
+Go back to the root of the project and create a new Swift package, copy the xcframework and generated swift files to it:
+
+```
+MySwiftPackage
+├── Sources
+│   └── MySwiftPackage
+│       └── SwiftBridgeCore.swift
+│       └── my_rust_lib.swift
+├── MyRustLib.b.xcframework
+└── Package.swift
+```
+
+Add the framework as a binary target
+```swift
+// MySwiftPackage/Package.swift
+
+let package = Package(
+    name: "MySwiftPackage",
+    products: [
+        .library(
+            name: "MySwiftPackge",
+            targets: ["MySwiftPackge"]),
+    ],
+    dependencies: [
+
+    ],
+    targets: [
+        .binaryTarget(
+            name: "MyRustLib",
+            path: "MyRustLib.xcframework"
+        ),
+        .target(
+            name: "MySwiftPackge",
+            dependencies: ["MyRustLib"]),
+    ]
+)
+```
+
+We will need to import our rust library in the `.swift` files:
```

Comment:
Why? Is there an alternative approach we can take instead of modifying our generated files?